Remove elastic index registry logic

joxl opened the pull request:
## Technical Summary

This registry module is no longer needed (has been obsolete ever since the document adapters were implemented). This PR is the predecessor to another (forthcoming) that will implement migration utilities for index management.

This change has added value now that both @AmitPhulera and I are both working on the Elasticsearch upgrade task because it reduces extra complexity for his work.

One huge "byproduct" gain of this change is that the `es_test` decorator now implements the Elastic index setup/teardown logic which I've been meaning to add for a long time. There is a lot of discrete Elastic index setup/teardown logic in tests that can now be ripped out when we have the time for that cleanup.

Review commit by commit.

Jira: [SAAS-14003](https://dimagi-dev.atlassian.net/browse/SAAS-14003)

## Safety Assurance

### Safety story

Change is fully covered by tests. Majority of changes _are_ test code. Production code changes are safe because their function is already superseded by the adapter layer.

### Automated test coverage

Covered by automated tests.

### QA Plan

Invisible, no QA.

### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change

joxl committed:
Remove `corehq.apps.es.registry`

joxl committed:
Overhaul `es_test` decorator

- now supports index setup/teardown boilerplate
- tests passing again

joxl committed:
Fix pillows mappings test

joxl committed:
Fix pillowtop tests

joxl committed:
Fix util tests and remove Elastic index setup boilerplate

AmitPhulera commented on corehq/ex-submodules/pillowtop/tests/test_bulk.py:
```diff
@@ -52,7 +52,7 @@ def test_get_errors_with_ids(self):
 
 @sharded
-@es_test(index=TEST_INDEX_INFO)
+@es_test
```
What does this decorator do in these tests?

joxl replied:
It adds the `es_test` nose attribute, so you can run all decorated tests like this:

```shell
./manage.py test --attr=es_test
```

AmitPhulera replied:
Ohh! Got it. Thanks 👍

snopoke reviewed (approved):
Nice cleanup. A few small comments but nothing blocking.

joxl committed:
Fix `SimpleTestCase` abuse in meta test tests

joxl committed:
Put previously fixed tests inside their own test case

joxl committed:
Use `assert_raises_regex` as suggested by @snopoke

joxl commented on corehq/apps/es/tests/test_test_utils.py:
```diff
@@ -1,180 +1,153 @@
 
+        assert False, "ValueError not raised"
+    except ValueError as exc:
+        assert str(exc).startswith("'setup_class' expects a classmethod"), str(exc)
```
See 21d93912

AmitPhulera reviewed (approved):
⭐

AmitPhulera commented:
I just merged the reindex PR which resulted in some conflicts with this branch.

joxl committed:
Merge branch 'master' into jm/rm-elastic-registry

joxl committed:
Remove registry from new index sync command
Co-authored-by: Amit Phulera <amitphulera999@gmail.com>

- update tests to use `es_test` for index setup/teardown
- speed up tests by 20 seconds (remove Elastic task poll delay)
- remove test dependency on private global dict
